azavea/echo-locator
# Filter listings by neighborhood max rent

## Overview

This filters both BHA and Realtor listings according to the max budget, defined by the number of bedrooms and neighborhood chosen, displayed within the neighborhood details sidebar.

### Notes

I originally created a "get the budget" function in the `actions/listings.js` file that would call the utility function `getNeighborhoodById()` so that the correct neighborhood's properties could be referenced to get the max rent and populate the API query. However, since that logic is a reproduction of the `detailNeighborhood` selector and the resulting budget value is also used to display the max rent in the sidebar, it felt worthwhile to instead create a separate `estMaxRent` selector that could be applied broadly. I then moved the logic to build the query outside of the fetch listings action so that `getState()` doesn't need to be called and there is less repeatable code.


## Testing Instructions

 * Run `./scripts/server`
 * Log in and select a neighborhood
 * Confirm correct max rent displays in sidebar and used in links (Zillow, Craigslist, Hotpads, GoSection8) as well as links section header — neighborhood max rent can be found in `neighborhoods.json` or by printing `neighborhood.properties`
 * Show Realtor and BHA listings and confirm no listing exceeds the max rent displayed in sidebar 


Resolves #402
